app/assets/javascripts/repositories/show.js

```diff
@@ -197,8 +197,29 @@
     });
   }
 
+  function initFilterSaving() {
```

Comment:
Maybe try without form? This looks overcomplicated, maybe just 2 independent ajax requests?